backup/src/test/java/io/camunda/zeebe/backup/processing/MockProcessingResult.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.backup.processing;
+
+import io.camunda.zeebe.engine.api.PostCommitTask;
+import io.camunda.zeebe.engine.api.ProcessingResult;
+import io.camunda.zeebe.engine.api.ProcessingResultBuilder;
+import io.camunda.zeebe.engine.processing.streamprocessor.writers.CommandResponseWriter;
+import io.camunda.zeebe.logstreams.log.LogStreamBatchWriter;
+import io.camunda.zeebe.msgpack.UnpackedObject;
+import io.camunda.zeebe.protocol.record.RecordType;
+import io.camunda.zeebe.protocol.record.RecordValue;
+import io.camunda.zeebe.protocol.record.RejectionType;
+import io.camunda.zeebe.protocol.record.ValueType;
+import io.camunda.zeebe.protocol.record.intent.Intent;
+import java.util.ArrayList;
+import java.util.List;
+
+record MockProcessingResult(List<Event> records) implements ProcessingResult {
+
+  @Override
+  public long writeRecordsToStream(final LogStreamBatchWriter logStreamBatchWriter) {
+    return 0;
+  }
+
+  @Override
+  public boolean writeResponse(final CommandResponseWriter commandResponseWriter) {
+    return false;
+  }
+
+  @Override
+  public boolean executePostCommitTasks() {
+    return false;
+  }
+
+  record Event(
```

Comment:
:wrench: I think we could do with just building `TypedRecordImpl` instances here, wrapped over a mocked `LoggedEvent` like here: https://github.com/camunda/zeebe/blob/087bf0e7240c873e28685aedfc46903a805d911b/engine/src/test/java/io/camunda/zeebe/engine/state/instance/BlackListStateTest.java#L93-L121

This would then also allow use to use `RecordAssert` in the tests.

Reply:
I would try to limit the depedency to `streamprocessor` as much as possible. Even though now it is in the same package as the `RecordProcessor` api, I assume we want to eventually split into a different package. Then the backup should only depend on the api, not on `StreamProcessor`.